undefined symbol: MUTABLE_HV with perl-5.10.0

Discovered this today when updating the modules in my new Solaris smoker and had to revert the version.pm for both threaded and unthreaded v5.10.0 back to 0.9924

```
azkaban ~/build/version-0.9927 $ perl -Iblib/lib -Iblib/arch -Mversion -e 'print version->new("0.99")'
perl: symbol lookup error: blib/arch/auto/version/vxs/vxs.so: undefined symbol: MUTABLE_HV
azkaban ~/build/version-0.9927 $ 
```

0.99_25 is fine, 0.99_26 also fails in a similar manner.

Oh not just on Solaris, the above is Linux.

Full output from test build:
```
[root@canker /opt/build]# PERL5_CPANPLUS_HOME=/opt/perl-5.10.0 /opt/perl-5.10.0/bin/perl tools/installer.pl --test LEONT/version-0.9927 
[MSG] Trying to get 'http://cpan.mirror.local/CPAN/authors/id/L/LE/LEONT/version-0.9927.tar.gz'
[MSG] Using cached file '/opt/perl-5.10.0/.cpanplus/authors/id/L/LE/LEONT/CHECKSUMS' on disk; ttl (3600s) is not exceeded
[MSG] Checksum matches for 'version-0.9927.tar.gz'
[MSG] Extracting 'version-0.9927.tar.gz'
[MSG] Extracted 'version' to '/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927'
[MSG] CPANPLUS is prefering Build.PL
[MSG] Loading YACSmoke database "/opt/perl-5.10.0/.cpanplus/cpansmoke.dat"
[MSG] Module 'version' depends on 'ExtUtils::MakeMaker', may need to build a 'CPANPLUS::Dist::YACSmoke' package for it as well
Running [/opt/perl-5.10.0/bin/perl -MCPANPLUS::Internals::Utils::Autoflush /opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/Makefile.PL]...
Testing if you have a C compiler
Checking if your kit is complete...
Looks good
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for version
Writing MYMETA.yml and MYMETA.json
[MSG] Checking for previous PASS result for "version-0.9927"
[MSG] Module 'version' depends on 'ExtUtils::MakeMaker', may need to build a 'CPANPLUS::Dist::YACSmoke' package for it as well
[MSG] Module 'version' depends on 'File::Temp', may need to build a 'CPANPLUS::Dist::YACSmoke' package for it as well
[MSG] Module 'version' depends on 'Test::More', may need to build a 'CPANPLUS::Dist::YACSmoke' package for it as well
[MSG] Module 'version' depends on 'base', may need to build a 'CPANPLUS::Dist::YACSmoke' package for it as well
Running [/usr/bin/make UNINST=1]...
Running Mkbootstrap for vxs ()
chmod 644 "vxs.bs"
"/opt/perl-5.10.0/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- vxs.bs ../blib/arch/auto/version/vxs/vxs.bs 644
"/opt/perl-5.10.0/bin/perl" "/opt/perl-5.10.0/lib/5.10.0/ExtUtils/xsubpp"  -typemap '/opt/perl-5.10.0/lib/5.10.0/ExtUtils/typemap'  vxs.xs > vxs.xsc
mv vxs.xsc vxs.c
cc -c    -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -I/usr/local/include -O2    -DVERSION=\"0.9927\"  -DXS_VERSION=\"0.9927\" -DPIC -fPIC "-I/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE"   vxs.c
In file included from vxs.xs:15:
./vutil.c:303:18: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
        (void)hv_stores(MUTABLE_HV(hv), "qv", newSViv(qv));
                        ^
./vutil.c:303:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "qv", newSViv(qv));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:305:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "alpha", newSViv(alpha));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:307:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "width", newSViv(width));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:429:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "original", orig);
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:430:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "vinf", newSViv(1));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:438:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "original", orig);
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:441:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "original", newSVpvs("0"));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:446:11: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    (void)hv_stores(MUTABLE_HV(hv), "version", newRV_noinc(MUTABLE_SV(av)));
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:494:17: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
        if ( hv_exists(MUTABLE_HV(ver), "qv", 2) )
                       ^
./vutil.c:494:7: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        if ( hv_exists(MUTABLE_HV(ver), "qv", 2) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/hv.h:439:24: note: expanded from macro 'hv_exists'
    (hv_common_key_len((urkk), (zamm), (clunk), HV_FETCH_ISEXISTS, NULL, 0) \
                       ^~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:708:54: note: passing argument to parameter 'hv' here
PERL_CALLCONV void*     Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32, const int action, SV *val, const U32 hash)
                                                         ^
In file included from vxs.xs:15:
./vutil.c:495:12: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
            (void)hv_stores(MUTABLE_HV(hv), "qv", newSViv(1));
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:497:7: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        if ( hv_exists(MUTABLE_HV(ver), "alpha", 5) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/hv.h:439:24: note: expanded from macro 'hv_exists'
    (hv_common_key_len((urkk), (zamm), (clunk), HV_FETCH_ISEXISTS, NULL, 0) \
                       ^~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:708:54: note: passing argument to parameter 'hv' here
PERL_CALLCONV void*     Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32, const int action, SV *val, const U32 hash)
                                                         ^
In file included from vxs.xs:15:
./vutil.c:498:12: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
            (void)hv_stores(MUTABLE_HV(hv), "alpha", newSViv(1));
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:500:18: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
            SV ** svp = hv_fetchs(MUTABLE_HV(ver), "width", FALSE);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:503:9: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
                (void)hv_stores(MUTABLE_HV(hv), "width", newSViv(width));
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:507:18: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
            SV ** svp = hv_fetchs(MUTABLE_HV(ver), "original", FALSE);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:509:9: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
                (void)hv_stores(MUTABLE_HV(hv), "original", newSVsv(*svp));
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:511:8: warning: implicit declaration of function 'MUTABLE_AV' is invalid in C99 [-Wimplicit-function-declaration]
        sav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(ver), "version", FALSE)));
              ^
./vutil.c:511:25: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        sav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(ver), "version", FALSE)));
                         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/sv.h:1327:24: note: expanded from macro 'SvRV'
#    define SvRV(sv) ((sv)->sv_u.svu_rv)
                       ^~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:511:6: warning: incompatible integer to pointer conversion assigning to 'AV *' (aka 'struct av *') from 'int' [-Wint-conversion]
        sav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(ver), "version", FALSE)));
            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./vutil.c:520:8: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        (void)hv_stores(MUTABLE_HV(hv), "version", newRV_noinc(MUTABLE_SV(av)));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:299:32: note: expanded from macro 'hv_stores'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:635:29: warning: expression result unused [-Wunused-value]
            LC_NUMERIC_LOCK(0);    /* Start critical section */
                            ^
./vutil.c:869:23: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
         && (svp = hv_fetchs(MUTABLE_HV(vs), "version", FALSE))
                             ^
./vutil.c:869:13: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
         && (svp = hv_fetchs(MUTABLE_HV(vs), "version", FALSE))
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:914:20: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
    if ( hv_exists(MUTABLE_HV(vs), "alpha", 5 ) )
                   ^
./vutil.c:914:10: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    if ( hv_exists(MUTABLE_HV(vs), "alpha", 5 ) )
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/hv.h:439:24: note: expanded from macro 'hv_exists'
    (hv_common_key_len((urkk), (zamm), (clunk), HV_FETCH_ISEXISTS, NULL, 0) \
                       ^~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:708:54: note: passing argument to parameter 'hv' here
PERL_CALLCONV void*     Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32, const int action, SV *val, const U32 hash)
                                                         ^
In file included from vxs.xs:15:
./vutil.c:923:17: warning: implicit declaration of function 'MUTABLE_AV' is invalid in C99 [-Wimplicit-function-declaration]
    if ( !(av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE))) ) ) {
                ^
./vutil.c:923:34: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    if ( !(av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE))) ) ) {
                           ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/sv.h:1327:24: note: expanded from macro 'SvRV'
#    define SvRV(sv) ((sv)->sv_u.svu_rv)
                       ^~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:923:15: warning: incompatible integer to pointer conversion assigning to 'AV *' (aka 'struct av *') from 'int' [-Wint-conversion]
    if ( !(av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE))) ) ) {
              ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./vutil.c:985:10: warning: implicit declaration of function 'MUTABLE_AV' is invalid in C99 [-Wimplicit-function-declaration]
    av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE)));
         ^
./vutil.c:985:37: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
    av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE)));
                                    ^
./vutil.c:985:27: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE)));
                    ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/sv.h:1327:24: note: expanded from macro 'SvRV'
#    define SvRV(sv) ((sv)->sv_u.svu_rv)
                       ^~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:985:8: warning: incompatible integer to pointer conversion assigning to 'AV *' (aka 'struct av *') from 'int' [-Wint-conversion]
    av = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(vs), "version", FALSE)));
       ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./vutil.c:1038:21: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
    svp = hv_fetchs(MUTABLE_HV(vs), "original", FALSE);
                    ^
./vutil.c:1038:11: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    svp = hv_fetchs(MUTABLE_HV(vs), "original", FALSE);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:1052:7: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        if ( hv_exists(MUTABLE_HV(vs), "qv", 2) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/hv.h:439:24: note: expanded from macro 'hv_exists'
    (hv_common_key_len((urkk), (zamm), (clunk), HV_FETCH_ISEXISTS, NULL, 0) \
                       ^~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:708:54: note: passing argument to parameter 'hv' here
PERL_CALLCONV void*     Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32, const int action, SV *val, const U32 hash)
                                                         ^
In file included from vxs.xs:15:
./vutil.c:1090:11: warning: implicit declaration of function 'MUTABLE_AV' is invalid in C99 [-Wimplicit-function-declaration]
    lav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(lhv), "version", FALSE)));
          ^
./vutil.c:1090:38: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
    lav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(lhv), "version", FALSE)));
                                     ^
./vutil.c:1090:28: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    lav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(lhv), "version", FALSE)));
                     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/sv.h:1327:24: note: expanded from macro 'SvRV'
#    define SvRV(sv) ((sv)->sv_u.svu_rv)
                       ^~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:1090:9: warning: incompatible integer to pointer conversion assigning to 'AV *' (aka 'struct av *') from 'int' [-Wint-conversion]
    lav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(lhv), "version", FALSE)));
        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./vutil.c:1093:28: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
    rav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(rhv), "version", FALSE)));
                     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/handy.h:294:32: note: expanded from macro 'hv_fetchs'
  ((SV **)Perl_hv_common(aTHX_ (hv), NULL, STR_WITH_LEN(key), 0,        \
                               ^
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/sv.h:1327:24: note: expanded from macro 'SvRV'
#    define SvRV(sv) ((sv)->sv_u.svu_rv)
                       ^~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:707:46: note: passing argument to parameter 'tb' here
PERL_CALLCONV void*     Perl_hv_common(pTHX_ HV* tb, SV* keysv, const char* key, STRLEN klen, int flags, int action, SV* val, U32 hash);
                                                 ^
In file included from vxs.xs:15:
./vutil.c:1093:9: warning: incompatible integer to pointer conversion assigning to 'AV *' (aka 'struct av *') from 'int' [-Wint-conversion]
    rav = MUTABLE_AV(SvRV(*hv_fetchs(MUTABLE_HV(rhv), "version", FALSE)));
        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from vxs.xs:23:
./vxs.inc:161:21: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
            if ( hv_exists(MUTABLE_HV(SvRV(req)), "qv", 2 ) ) {
                           ^
./vxs.inc:161:11: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
            if ( hv_exists(MUTABLE_HV(SvRV(req)), "qv", 2 ) ) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/hv.h:439:24: note: expanded from macro 'hv_exists'
    (hv_common_key_len((urkk), (zamm), (clunk), HV_FETCH_ISEXISTS, NULL, 0) \
                       ^~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:708:54: note: passing argument to parameter 'hv' here
PERL_CALLCONV void*     Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32, const int action, SV *val, const U32 hash)
                                                         ^
In file included from vxs.xs:23:
./vxs.inc:377:17: warning: implicit declaration of function 'MUTABLE_HV' is invalid in C99 [-Wimplicit-function-declaration]
        if ( hv_exists(MUTABLE_HV(lobj), key, keylen ) )
                       ^
./vxs.inc:377:7: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'HV *' (aka 'struct hv *') [-Wint-conversion]
        if ( hv_exists(MUTABLE_HV(lobj), key, keylen ) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/hv.h:439:24: note: expanded from macro 'hv_exists'
    (hv_common_key_len((urkk), (zamm), (clunk), HV_FETCH_ISEXISTS, NULL, 0) \
                       ^~~~~~
/opt/perl-5.10.0/lib/5.10.0/amd64-freebsd/CORE/proto.h:708:54: note: passing argument to parameter 'hv' here
PERL_CALLCONV void*     Perl_hv_common_key_len(pTHX_ HV *hv, const char *key, I32 klen_i32, const int action, SV *val, const U32 hash)
                                                         ^
47 warnings generated.
rm -f ../blib/arch/auto/version/vxs/vxs.so
cc  -shared  -L/usr/local/lib  vxs.o  -o ../blib/arch/auto/version/vxs/vxs.so        
chmod 755 ../blib/arch/auto/version/vxs/vxs.so
Manifying 2 pod documents
Running [/usr/bin/make test UNINST=1]...
"/opt/perl-5.10.0/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- vxs.bs ../blib/arch/auto/version/vxs/vxs.bs 644
"/opt/perl-5.10.0/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- vxs.bs ../blib/arch/auto/version/vxs/vxs.bs 644
No tests defined for version::vxs extension.
PERL_DL_NONLAZY=1 "/opt/perl-5.10.0/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00impl-pp.t ..... ok
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/01base.t ........ 
Dubious, test returned 1 (wstat 256, 0x100)
All 1 subtests passed 
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/02derived.t ..... 
Dubious, test returned 1 (wstat 256, 0x100)
All 1 subtests passed 
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/03require.t ..... 
Dubious, test returned 1 (wstat 256, 0x100)
All 4 subtests passed 
t/04strict_lax.t .. ok
t/05sigdie.t ...... ok
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/06noop.t ........ 
Dubious, test returned 1 (wstat 256, 0x100)
All 1 subtests passed 
t/07locale.t ...... ok
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/08_corelist.t ... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 2/3 subtests 
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/09_list_util.t .. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 2/3 subtests 
/opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927/blib/arch/auto/version/vxs/vxs.so: Undefined symbol "MUTABLE_HV"
t/10_lyon.t ....... 
Dubious, test returned 1 (wstat 256, 0x100)
No subtests run 
t/11_taint.t ...... skipped: No Test::Taint available

Test Summary Report
-------------------
t/01base.t      (Wstat: 256 Tests: 1 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
t/02derived.t   (Wstat: 256 Tests: 1 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
t/03require.t   (Wstat: 256 Tests: 4 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
t/06noop.t      (Wstat: 256 Tests: 1 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
t/07locale.t    (Wstat: 0 Tests: 8 Failed: 0)
  TODO passed:   7
t/08_corelist.t (Wstat: 256 Tests: 1 Failed: 0)
  Non-zero exit status: 1
  Parse errors: Bad plan.  You planned 3 tests but ran 1.
t/09_list_util.t (Wstat: 256 Tests: 1 Failed: 0)
  Non-zero exit status: 1
  Parse errors: Bad plan.  You planned 3 tests but ran 1.
t/10_lyon.t     (Wstat: 256 Tests: 0 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
Files=12, Tests=844,  2 wallclock secs ( 0.20 usr  0.05 sys +  2.02 cusr  0.26 csys =  2.52 CPU)
Result: FAIL
Failed 7/12 test programs. 0/844 subtests failed.
*** Error code 255

Stop.
make: stopped in /opt/perl-5.10.0/.cpanplus/5.10.0/build/pmlgLnzmBt/version-0.9927
[MSG] Sending test report for 'version-0.9927'
[MSG] Successfully sent 'fail' report for 'version-0.9927'
```

--- a/vutil/vutil.h
+++ b/vutil/vutil.h
@@ -223,7 +223,7 @@ const char * Perl_prescan_version(pTHX_ const char *s, bool strict, const char**
 
 
 #if PERL_VERSION_LT(5,27,9)
-#  define LC_NUMERIC_LOCK
+#  define LC_NUMERIC_LOCK(cond)
 #  define LC_NUMERIC_UNLOCK
 #  if PERL_VERSION_LT(5,19,0)
 #    undef STORE_LC_NUMERIC_SET_STANDARD

--- a/vutil/vutil.h
+++ b/vutil/vutil.h
@@ -16,12 +16,24 @@
 #  else
 #    define MUTABLE_PTR(p) ((void *) (p))
 #  endif
+#endif
 
+#ifndef MUTABLE_AV
 #  define MUTABLE_AV(p)	((AV *)MUTABLE_PTR(p))
+#endif
+#ifndef MUTABLE_CV
 #  define MUTABLE_CV(p)	((CV *)MUTABLE_PTR(p))
+#endif
+#ifndef MUTABLE_GV
 #  define MUTABLE_GV(p)	((GV *)MUTABLE_PTR(p))
+#endif
+#ifndef MUTABLE_HV
 #  define MUTABLE_HV(p)	((HV *)MUTABLE_PTR(p))
+#endif
+#ifndef MUTABLE_IO
 #  define MUTABLE_IO(p)	((IO *)MUTABLE_PTR(p))
+#endif
+#ifndef MUTABLE_SV
 #  define MUTABLE_SV(p)	((SV *)MUTABLE_PTR(p))
 #endif
 

